sdk/eventhub/event-hubs/src/sender.ts

```diff
@@ -107,13 +123,17 @@ export class EventHubProducer {
     options?: SendOptions
   ): Promise<void> {
     this._throwIfSenderOrConnectionClosed();
-    throwTypeErrorIfParameterMissing(this._context.connectionId, "eventData", eventData);
+    if (Array.isArray(eventData) && eventData.length === 0) {
+      log.error(`[${this._context.connectionId}] No events to send.`);
+      return;
+    }
     if (eventData instanceof EventDataBatch && !eventData.batchMessage) {
       log.error(
         `[${this._context.connectionId}] No events to send, use tryAdd() function on the EventDataBatch to add events in a batch.`
```

Comment:
```suggestion
        `[${this._context.connectionId}] Empty batch was passsed. No events to send.`
```

Reply:
Here I think we should suggest our users to use tryAdd() method?

`use tryAdd() function on the EventDataBatch to add events in a batch`

Reply:
When an empty batch is encountered, it is more likely that that the users code had no events to add and less likely that they didnt know how to add events.
Also, we are only logging this and not throwing the error. The logs need not have instructions on what to do.